tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.cpp

```diff
@@ -144,6 +145,11 @@ void EltwiseBinary::validate(const std::vector<Tensor>& input_tensors) const {
     TT_FATAL(
         (input_tensor_a.get_layout() == Layout::TILE && input_tensor_b.get_layout() == Layout::TILE),
         "Inputs to eltwise binary must be tilized");
+    if(!output_tensors.empty() && output_tensors.at(0).has_value()){
+        const auto output_shape_required = this->compute_output_shapes(input_tensors);
+        const auto& op_tensor = output_tensors.at(0).value();
+        TT_FATAL(op_tensor.get_legacy_shape() == output_shape_required.at(0), fmt::format("The input tensors need a shape of {}, however the output tensor is only {}", output_shape_required,  op_tensor.get_legacy_shape()));
+    }
```

Comment:
The checks against `this->output_mem_config` in the if statement `if (input_tensor_a.memory_config().is_sharded()) {`
need to use the output tensor's mem config if output tensor was provided instead of `this->output_mem_config`

Reply:
ok. I've made changes to check output_tensor's memory_config() if output tensor is passed , or else continue with existing check `this->output_mem_config`